### DISCUSSION TITLE
Pyodide Success Story

### DISCUSSION THREAD

**Discussion by alynch4047** [pyodide/pyodide#1907] [Show and tell]
Hi all I just wanted to share how Pyodide has really benefited me. I have a medium sized python project that parses a fairly complex piece of json data in order to construct another complex data structure. This functionality needs to be accessed from a Chrome extension and is (was) being served via a flask server. I've been thinking about how to distribute the python based functionality and had created a docker image for that, when I realised I could try using Pyodide to actually host and run the python from inside the browser/extension (not as a server but directly accessing the api). It took a couple of days fighting with Chrome extensions and security policies (had to drop back to Manifest v2 to allow the web assembly to run, this is a known issue in Chrome extensions CSP stuff) but got Pyodide to run my parser within the browser extension! Even better, it subjectively seems to run as fast as the externally hosted version. The whole piece of functionality is now entirely provided and hosted via the extension itself. Thanks to all who contributed to Pyodide.

**Comment by rth** [comment]
Thanks for sharing, @alynch4047 ! Good to hear that!

> It took a couple of days fighting with Chrome extensions and security policies

Actually it might be helpful to share the process. I also wanted to explore at some point how to do a browser extension  with Pyodide and that part seemed tedious. People would likely be interested in it. We likely can't put this in the docs, as things on the browser side change occasionally and it would take effort to keep up to date. However maybe this could work as a blog post?

We were considering to create a Pyodide blog https://github.com/pyodide/pyodide/issues/1766. In case you are interested in writing a blog post about your experience (I can help with editing/review), we can certainly move forward with that. Or if you prefer another platform that's fine too.

**Reply by rth** [reply]
Actually I saw that @grimmer0125 also made a Chrome extension. So same question to you if you are interesed. Or it could be a shared blog post. We are looking for material :)

**Reply by grimmerk** [reply]
I will be happy that I can share my experience about how to migrate the existing one to Pyodide base. Actually, I'm thinking to restart writing my blog and are looking for a good blog platform. However, writing one on Pyodide's future blog does not have a conflict with writing one on any blog I own. Also, my written plan may be written in Traditional Chinese (used in Taiwan) or possibly will not happen (maintaining a blog takes time). Just inform me when Pyodide's blog platform is ready. 

Actually, I have shared this experience in PyCon Taiwan 2021 and will share the same talk in the coming PyCon APAC 2021 conference. Thanks for the great project, Pyodide.

**Reply by rth** [reply]
Great @grimmer0125 and thanks for talking about Pyodide at conferences! If you write it in Traditional Chinese for the personal blog and then translate it in English for our blog that would definitely work.

**Reply by grimmerk** [reply]
I can give the material after finishing the conference talk.

**Reply by rth** [reply]
That works. Please open an issue about it at https://github.com/pyodide/pyodide-blog with a brief summary of the proposed post. Thanks!
